pkgs/tools/backup/restic/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, lib, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "restic-${version}";
+  version = "0.5.0";
+
+  goPackagePath = "github.com/restic/restic";
+
+  src = fetchFromGitHub {
+    owner = "restic";
+    repo = "restic";
+    rev = "v${version}";
+    sha256 = "0dj6zg4b00pwgs6nj7w5s0jxm6cfavd9kdcq0z4spypwdf211cgl";
+  };
+
+  buildPhase = ''
```

Comment:
Did you tried to build it without default `buildPhase`/`installPhase`? It doesn't look like it needs custom ones.